### PR TITLE
DRIVERS-2416 Add docs and tests for azure oidc

### PR DESCRIPTION
**Pull request by blink1073**
Tested with https://github.com/mongodb/mongo-python-driver/pull/1443 

- [x] Passing build in Python PR
	`======================== 15 passed, 21 skipped in 3.69s ========================`
- [x] Adjust EVG tasks so OIDC is grouped and Docker test is a new top level test
- [x] Update Azure setup wiki and update link in readme
- [x] Set up VMs to automatically shut down as a failsafe
- [x] See if we can have a default client id on the VM and if so, add a test for it - no, client_id is [required](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http) when there is more than on managed id.

### CODE REVIEW

**Comment by blink1073** [comment]
evergreen retry

**Comment by blink1073** [comment]
@matthewdale I plan to update the top level `auth_oidc` readme in a subsequent PR.

**Comment by blink1073** [comment]
evergreen retry

**Review by durran — CHANGES_REQUESTED** [review]

File: .evergreen/auth_oidc/azure/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+TOKEN_CLIENT_ID=os.environ["AZUREOIDC_TOKENCLIENT"]  # For first user
+TOKEN_CLIENT_ID=os.environ["AZUREOIDC_TOKENCLIENT2"]  # For second user
+```
+
+Note: If you are creating a uri, you will have to escape `TOKEN_AUDIENCE` value, e.g.
+
+```bash
+MONGODB_URI="${MONGODB_URI}/?authMechanism=MONGODB-OIDC"
+MONGODB_URI="${MONGODB_URI}&authMechanismProperties=PROVIDER_NAME:azure"
+MONGODB_URI="${MONGODB_URI},TOKEN_AUDIENCE:api%3A%2F%2F${AZUREOIDC_CLIENTID}"
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/azure/README.md:72]: There are two different clients though, so in the Python tests we add them as MongoClient options.

[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/azure/README.md:72]: Added a note.

[the remark this answers is not in the document]
Comment (durran) [reply .evergreen/auth_oidc/azure/README.md:72]: Thanks.

**Review by durran — APPROVED** [review]

**Pull request merged**